replication-api-impl/src/test/java/com/connexta/replication/api/impl/worker/WorkerManagerComponentTest.java

```diff
@@ -118,43 +151,139 @@ public void cleanup() throws Exception {
-    queue.put(taskInfo2);
-    queue.put(taskInfo3);
+    TaskInfo taskInfo = createTaskInfo(METADATA_ID, metadataInfo, resourceInfo);
+    Task task = queue.putAndPeek(taskInfo);
 
+    ArgumentCaptor<CreateStorageRequest> createStorageRequestCaptor = ArgumentCaptor
+        .forClass(CreateStorageRequest.class);
+    when(localAdapter.createResource(createStorageRequestCaptor.capture())).thenReturn(true);
+
+    Resource resource = mock(Resource.class);
+    ResourceResponse resourceResponse = mock(ResourceResponse.class);
+    when(resourceResponse.getResource()).thenReturn(resource);
+
+    ArgumentCaptor<ResourceRequest> resourceRequestCaptor = ArgumentCaptor
+        .forClass(ResourceRequest.class);
+    when(remoteAdapter.readResource(resourceRequestCaptor.capture())).thenReturn(resourceResponse);
+
+    // when
     workerManager.init();
 
-    assertTimeoutPreemptively(
-        Duration.ofSeconds(10),
-        () -> {
-          while (true) {
-            if (queue.size() == 0) {
-              break;
-            }
-          }
-        });
+    // then
+    queue.waitForSizeToReach(0, 10, TimeUnit.SECONDS);
+
+    assertTrue(task.isCompleted());
+    ResourceRequest sentResourceRequest = resourceRequestCaptor.getValue();
+
+    final Metadata sentMetadata = sentResourceRequest.getMetadata();
+    assertEquals(metadata.getType(), sentMetadata.getType());
+    assertEquals(metadata.getMetadataModified(), sentMetadata.getMetadataModified());
+    assertEquals(metadata.getRawMetadata(), sentMetadata.getRawMetadata());
+    assertEquals(metadata.getId(), sentMetadata.getId());
   }
 
-  private SitePojo createSitePojo(String id, int seed) {
+  @Test
+  public void testWorkerHarvestDdfMetadata() throws Exception {
+    final MemorySiteQueue queue = (MemorySiteQueue) broker.getQueue(REMOTE_SITE_ID);
+    // for DDF metadata, it is assumed Task id == metadata ID
+    final Metadata metadata = createMetadata(METADATA_ID);
+    final TaskInfo taskInfo = createTaskInfo(METADATA_ID, createMetadataInfo(metadata));
+    final Task task = queue.putAndPeek(taskInfo);
+
+    ArgumentCaptor<CreateRequest> createRequestCaptor = ArgumentCaptor
+        .forClass(CreateRequest.class);
+    when(localAdapter.createRequest(createRequestCaptor.capture())).thenReturn(true);
+
+    // when
+    workerManager.init();
+
+    // then
+    queue.waitForSizeToReach(0, 10, TimeUnit.SECONDS);
+    assertTrue(task.isCompleted());
+    assertFalse(task.isLocked());
+
+    CreateRequest createRequest = createRequestCaptor.getValue();
+    Metadata sentMetadata = createRequest.getMetadata().get(0);
+    assertEquals(metadata.getType(), sentMetadata.getType());
+    assertEquals(metadata.getMetadataModified(), sentMetadata.getMetadataModified());
+    assertEquals(metadata.getRawMetadata(), sentMetadata.getRawMetadata());
+    assertEquals(metadata.getId(), sentMetadata.getId());
+  }
+
+  @Test
+  public void testMultipleTasksCompleted() throws Exception {
```

Comment:
✏️ It be nice to test with one task that you know you will ignore and unlock and one task that fails

Reply:
I did this test mostly to check that a worker will continue to take from the queue after processing 1. I can add separate tests for an ignored and failed task though.

Reply:
I didn't mean in this test. For sure additional tests that verifies that would be great.